### PR TITLE
rename context.Query to context.Form

### PR DESCRIPTION
**Pull request by lunny**
The name `Query` on context derived from macaron is confusing because it should only get query parameters from URL query.
This PR renames `Query` to `Form` as the first step. Next step, we need add the real `Query` methods and change some routers' revocations.

### CODE REVIEW

**Comment by coolaj86** [comment]
> we need add the real `Query` methods

Why?

`ctx.Req.URL.Query()` is easy to understand without having to be familiar with a particular framework.

Why add layers on top of native Go APIs that are already simple, easy to use, well-understood, with predictable (and desirable) behavior?

**Review by lafriks — APPROVED** [review]

File: modules/context/context.go [hunk 1]
```diff
@@ -287,41 +287,38 @@ func (ctx *Context) Header() http.Header {
...
-// Currently to be compatible with macaron, we keep it.
-
-// Query returns request form as string with default
-func (ctx *Context) Query(key string, defaults ...string) string {
+// Form returns request form as string with default
+func (ctx *Context) Form(key string, defaults ...string) string {
 	return (*Forms)(ctx.Req).MustString(key, defaults...)
 }
 
 // QueryTrim returns request form as string with default and trimmed spaces
```
Comment (lafriks) [inline modules/context/context.go]: Please fix comments

Comment (lafriks) [reply modules/context/context.go]:
```suggestion
// FormTrim returns request form as string with default and trimmed spaces
```

**Review by 6543 — CHANGES_REQUESTED** [review]
I thought it will already work as named ?!?

**Comment by lunny** [comment]
> > Next step, we need add the real `Query` methods
> 
> Why?
> 
> `ctx.Req.URL.Query()` is easy to understand without having to be familiar with a particular framework.
> 
> Why add layers on top of native Go APIs that are already simple, easy to use, well-understood, well-documented, and that have predictable (and desirable) behavior?
> 
> Wouldn't it be better to go the opposite direction and remove the legacy Gogs/Macaron/Gitea code in favor of cleaner, simpler, native Go code?
> (unless there's a particular missing feature or unexpected caveat in how the native Go code works, of course)
> 
> I would much prefer to use the standard Go and Chi approaches:
> 
> ```go
> str := r.URL.Query().Get(k)
> val := r.FormValue(k)
> ctx := r.Context()
> data := r.Context().Value("data").(*Data)
> tpl := data["something"]
> ```
> 
> I also understand having a few convenience functions or perhaps an object to hold `Data`, as it is presently:
> 
> ```go
> r := ctx.Req
> str := r.URL.Query().Get(k)
> val := r.FormValue(k)
> tpl := ctx.Data["something"]
> ```
> 
> But the idea of adding new APIs that need new documentation around APIs that already work exactly as expected... makes me sad.

These simple wrap functions will help to convert string to int, bool or others. i.e. `QueryInt` will try to convert `r.URL.Query().Get(k)` to int. Since they will be used many places. You have to write `strconv.Atoi` and above codes everywhere.

**Comment by lunny** [comment]
> I thought it will already work as named ?!?

Like @coolaj86  QueryXXX should warp `r.URL.Query().Get(k)` but not `r.Form.Values`. So rename `Query` to `Form` to make less confusing.

File: modules/context/context.go (hunk 1 above)
Comment (lunny) [reply modules/context/context.go]: done.

**Comment by 6543** [comment]
This renamed funct. Are used in api a lot so we should migrate them all to use query directly. What porpuse does this form functions do have beside getting query values?!?

@lunny

**Comment by 6543** [comment]
Instead of renaming i propose to refactor it to do what the name sound like or just delete it

**Comment by 6543** [comment]
@lunny i would go this way: #16567

if you are ok with it, I'll finish my pull

**Comment by lunny** [comment]
> @lunny i would go this way: #16567
> 
> if you are ok with it, I'll finish my pull

I'm strongly against that PR because it's wrong.

I agree we need use QueryX for those GET requests, but to make review easier, I would like to make it as two steps. This is the first step to rename QueryX to FormX, next step to add QueryX back and move functions on GET requests to use QueryX .

**Review by 6543 — APPROVED** [review]
one step at a time ...

**Pull request merged**